src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -27,14 +27,26 @@ public partial class SdkDirectoryWorkloadManifestProvider : IWorkloadManifestPro
         private WorkloadSet? _manifestsFromInstallState;
         private string? _installStateFilePath;
 
+        //  This will be non-null if there is an error loading manifests that should be thrown when they need to be accessed.
+        //  We delay throwing the error so that in the case where global.json specifies a workload set that isn't installed,
+        //  we can successfully construct a resolver and install that workload set
+        private Exception? _exceptionToThrow = null;
+        string? _globalJsonWorkloadSetVersion;
+
         public SdkDirectoryWorkloadManifestProvider(string sdkRootPath, string sdkVersion, string? userProfileDir, string? globalJsonPath)
             : this(sdkRootPath, sdkVersion, Environment.GetEnvironmentVariable, userProfileDir, globalJsonPath)
         {
         }
 
         public static SdkDirectoryWorkloadManifestProvider ForWorkloadSet(string sdkRootPath, string sdkVersion, string? userProfileDir, string workloadSetVersion)
         {
-            return new SdkDirectoryWorkloadManifestProvider(sdkRootPath, sdkVersion, Environment.GetEnvironmentVariable, userProfileDir, globalJsonPath: null, workloadSetVersion);
+            return new SdkDirectoryWorkloadManifestProvider(
+                sdkRootPath,
+                sdkVersion,
+                Environment.GetEnvironmentVariable,
+                userProfileDir,
+                globalJsonPath: workloadSetVersion is null ? GetGlobalJsonPath(Environment.CurrentDirectory) : null,
```

Comment:
```suggestion
                globalJsonPath: null,
```

I don't think we should be getting the global.json path here if the passed in value is null.  If we do garbage collection within the context of a global.json file, we need to be able to create a resolver in the non-global.json global install state context, so we know which workload sets and manifests to keep.

If we resolve the global.json path here, then I think we'll never have access to that global state when running a garbage collection from a folder that has a global.json.

Reply:
This is true, but conversely, we need to make sure we have a way to notice the workload set in a global.json because we don't want to uninstall a workload set that we _know_ is in use by the repository we're currently in so that we can keep a global workload set we might not even be using.

I was thinking we'd keep the one that matters for our current case since we know we can't be perfect regardless. I can make it find both, but I don't want to just revert this part as you're suggesting. If the GC does its thing properly, reverting this part would break pinning to an older version via global.json.

Reply:
I guess it doesn't matter in terms of this PR anyway because it just doesn't GC at all, so it shouldn't matter if I just revert this, but when we properly implement GC for workload sets, we should remember to make a separate resolver for the current directory's global.json if necessary.

Reply:
When we do GC we will have a list of all the global.json files that we know about, so we will look at all of the workload sets that they refer to and then create resolvers for those workload sets.  But we will probably pass the workload set version in directly instead of having it being read from global.json in that case (no need to create a bunch of different resolvers for a bunch of different global.json files that all refer to the same workload set).